src/templating/content.md

```diff
@@ -0,0 +1,654 @@
+
+You have to write YAML files to change even a single value — unlike `yq`.
+
+_Is there an alternative to Kustomize and `yq` that is flexible and structured?_
+
+## Writing YAML with code
+
+You could generate YAML programatically with code.
+
+And you don't even need to start from scratch.
+
+Kubernetes has several official libraries where you can create objects such as Deployments and Pod with code.
+
+> Please note that since Kubernetes uses an OpenAPI, you can autogenerate a library for your favourite language — if there isn't one already.
+
+As an example, let's have a look at how you can use Javascript to create a Pod and programatically change it.
+
+```js|title=pod.js
+const { Pod, Container } = require('kubernetes-models/v1')
+
+const pod = new Pod({
+  metadata: {
+    name: 'test-pod',
+  },
+  spec: {
+    containers: [
+      new Container({
+        name: 'test-container',
+        image: 'k8s.gcr.io/busybox',
+        env: [{ name: 'ENV', value: 'production' }],
+      }),
+    ],
+  },
+})
+
+console.log(JSON.stringify(pod, null, 2))
+```
+
+You can execute the script with the `node` binary:
+
+```terminal|command=1|title=bash
+node pod.js
+{
+  "metadata": {
+    "name": "test-pod"
+  },
+  "spec": {
+    "containers": [
+      {
+        "name": "test-container",
+        "image": "k8s.gcr.io/busybox",
+        "env": [
+          {
+            "name": "ENV",
+            "value": "production"
+          }
+        ]
+      }
+    ]
+  },
+  "apiVersion": "v1",
+  "kind": "Pod"
+}
+```
+
+The output is a Pod, but it's in JSON and not YAML.
+
+That shouldn't be a problem because:
+
+1. YAML is a superset of JSON. Any JSON file is also a valid YAML file
+1. The Kubernetes API receives the resource in JSON even if you write YAML files. `kubectl` serialises the YAML into JSON
+
+_What if you want to change the environment variable?_
+
+Since this is just code, you can use native constructs:
+
+```js|title=pod.js
+const { Pod, Container } = require('kubernetes-models/v1')
+
+const pod = new Pod({
+  metadata: {
+    name: 'test-pod',
+  },
+  spec: {
+    containers: [
+      new Container({
+        name: 'test-container',
+        image: 'k8s.gcr.io/busybox',
+        env: [{ name: 'ENV', value: process.env.ENV || 'production' }],
+      }),
+    ],
+  },
+})
+
+console.log(JSON.stringify(pod, null, 2))
+```
+
+The code above read the environment variable `ENV` and, if set, set the according value in the JSON.
+
+```terminal|command=1|title=bash
+ENV=dev node pod.js
+{
+  "metadata": {
+    "name": "test-pod"
+  },
+  "spec": {
+    "containers": [
+      {
+        "name": "test-container",
+        "image": "k8s.gcr.io/busybox",
+        "env": [
+          {
+            "name": "ENV",
+            "value": "dev"
+          }
+        ]
+      }
+    ]
+  },
+  "apiVersion": "v1",
+  "kind": "Pod"
+}
+```
+
```

Comment:
You could save the above output in a file named `pod.json` and then create the Pod in the cluster with kubectl:

```bash
kubectl apply -f pod.json
```